### PR TITLE
chapter_recurrent-neural-networks/rnn-scratch

### PR DESCRIPTION
**Pull request by zhuyuanxiang**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-887/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Comment by mli** [comment]
Job d2l-zh/PR-887/9 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

File: chapter_recurrent-neural-networks/rnn-scratch.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# 循环神经网络的从零开始实现
+# 从零开始实现循环神经网络
```
Comment (goldmermaid) [inline chapter_recurrent-neural-networks/rnn-scratch.md:1]: Hey @zhuyuanxiang , can you change this back? It needs to align with the previous chapters (e.g. "线性回归的从零开始实现"). Thanks!

File: chapter_recurrent-neural-networks/rnn-scratch.md [hunk 2]
```diff
@@ -168,15 +168,15 @@ def init_rnn_state(batch_size, num_hiddens):
     return (d2l.zeros((batch_size, num_hiddens)), )
 ```
 
-[**下面的`rnn`函数定义了如何在一个时间步计算隐藏状态和输出。**]请注意，循环神经网络模型通过最外层维度`inputs`循环，以便逐时间步更新小批量的隐藏状态`H`。此外，这里的激活函数使用$\tanh$函数。如 :numref:`sec_mlp` 所述，当元素在实数上均匀分布时，$\tanh$函数的平均值为0。
+[**下面的`rnn`函数定义了如何在一个时间步内计算隐藏状态和输出。**]请注意，循环神经网络模型通过`inputs`最外层的维度实现循环，以便逐时间步更新小批量数据的隐藏状态`H`。此外，这里使用$\tanh$函数作为激活函数。如 :numref:`sec_mlp` 所述，当元素在实数上满足均匀分布时，$\tanh$函数的平均值为0。
 
 ```{.python .input}
 def rnn(inputs, state, params):
-    # `inputs`的形状：(`时间步数量`, `批量大小`, `词表大小`)
+    # `inputs`的形状：(`时间步数量`，`批量大小`，`词表大小`)
```
Comment (goldmermaid) [inline chapter_recurrent-neural-networks/rnn-scratch.md:175]: Great catches! Thanks for being fastidious!

**Review by goldmermaid — APPROVED** [review]
Thanks @zhuyuanxiang !

**Comment by mli** [comment]
Job d2l-zh/PR-887/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

File: chapter_recurrent-neural-networks/rnn-scratch.md (hunk 1 above)
Comment (zhuyuanxiang) [reply chapter_recurrent-neural-networks/rnn-scratch.md]: OK. It's done.

**Comment by mli** [comment]
Job d2l-zh/PR-887/11 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-887/

**Pull request merged**